iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -249,6 +228,7 @@ class string
     /// @endcode
     // TruncateToCapacity_t is a compile time variable to distinguish between constructors
     // NOLINTNEXTLINE(hicpp-named-parameter, readability-named-parameter)
+    /// @todo Make this c'tor explicit to avoid implicit conversion from 'nullptr'
     string(TruncateToCapacity_t, const std::string& other) noexcept;
```

Comment:
There's a few places were this c'tor is used implicitly if `iox::cxx::string(iox::cxx::TruncateToCapacity, char* string)` is called e.g. in C binding. I can push the changes next week. 

A better way to avoid this would be probably be to introduce a method like:

```cpp
// Avoid any implicit conversion, as template are always preferred in overload resolution
template <typename T>
string(TruncateToCapacity_t, const T other) = delete;
```

Reply:
I think this is a good idea for this particular use case. Btw. `explicit` works only for constructors with one argument therefore you have to implement the `explicit` use case for two or more arguments with your suggestion - I think we should do it!